optaplanner-core/src/main/java/org/optaplanner/core/impl/domain/locator/Locator.java

```diff
@@ -21,22 +21,20 @@
 import java.util.Map;
 
 import org.optaplanner.core.api.domain.locator.PlanningId;
-import org.optaplanner.core.api.domain.solution.PlanningSolution;
 import org.optaplanner.core.impl.domain.solution.descriptor.SolutionDescriptor;
 import org.optaplanner.core.impl.score.director.ScoreDirector;
 
 /**
- * @param <Solution_> the solution type, the class with the {@link PlanningSolution} annotation
  * @see PlanningId
  * @see ScoreDirector#locateWorkingObject(Object)
  */
-public class Locator<Solution_> {
+public class Locator {
 
     private final LocationStrategyResolver locationStrategyResolver;
 
     private Map<Object, Object> idToWorkingObjectMap;
 
-    public Locator(SolutionDescriptor<Solution_> solutionDescriptor) {
+    public Locator(SolutionDescriptor<?> solutionDescriptor) {
```

Comment:
yes, but I dislike both:
- raw type gives intellij warning and can hide good compilations errors
- ? also doesn't give any guarantees that the generic type is consistent across all classes - and it can cause ackwards casts that cast Foo<?> to Foo<Solution_> if the Foo<?> leaks into other classes.